File: js/signaldock/tests/unit/security.test.js
const test = require('node:test');
const assert = require('node:assert/strict');
const crypto = require('node:crypto');
const { verifySignature } = require('../../src/core/security');

test('verifySignature validates digest with constant-time compare', () => {
  const payload = 'manifest:v1';
  const digest = crypto.createHash('sha256').update(payload).digest('hex');
  assert.equal(verifySignature(payload, digest, digest), true);
  assert.equal(verifySignature(payload, digest.slice(0, -1), digest), false);
});

File: ruby/opalcommand/tests/unit/security_test.rb
# frozen_string_literal: true

require_relative '../test_helper'
require 'digest'

class SecurityTest < Minitest::Test
  def test_verify_signature_matches_digest
    payload = 'manifest:v1'
    digest = Digest::SHA256.hexdigest(payload)
    assert OpalCommand::Core::Security.verify_signature(payload, digest, digest)
    refute OpalCommand::Core::Security.verify_signature(payload, digest[0..-2], digest)
  end
end
